clang/lib/Sema/Sema.cpp

```diff
@@ -1658,9 +1661,13 @@ class DeferredDiagnosticsEmitter
     if (auto *FD = dyn_cast<FunctionDecl>(D)) {
       ShouldEmitRootNode = S.getEmissionStatus(FD, /*Final=*/true) ==
                            Sema::FunctionEmissionStatus::Emitted;
+      RootReason = S.getEmissionReason(FD);
       checkFunc(SourceLocation(), FD);
-    } else
+    } else {
+      // Global VarDecls don't really have a reason, so set this to 'ALL'.
+      RootReason = Sema::DDR_ALL;
```

Comment:
This seems suspicious -- for instance, we have SYCL attributes that only diagnose in one mode or another and those attributes may be applied to variables.

Reply:
The difference here is that global VarDecls always have their diagnostics emitted I think, right?